### PR TITLE
Support workspaces in Repository

### PR DESCRIPTION
**Pull request by hnts**
Hi there!

To use workspaces with PipelineRun, Repository CRD requires to support workspaces.

This PR implements a way to use workspaces in Repository CRD.

The following example uses workspaces.

```
apiVersion: polling.tekton.dev/v1alpha1
kind: Repository
metadata:
  name: demo-app-repository
  namespace: tekton-polling-operator
spec:
  url: https://github.com/bigkevmcd/tekton-polling-operator.git
  ref: main
  frequency: 5m
  type: github
  pipelineRef:
    name: ci-pipeline
    namespace: tekton-pipelines
    params:
    - name: gitRevision
      expression: commit.sha
    - name: repoURL
      expression: repoURL
    workspaces:
      - name: git-source
        persistentVolumeClaim:
          claimName: tekton-volume
```

### CODE REVIEW

**Comment by bigkevmcd** [comment]
Hi @hnts thanks for this, I've been holding off on adding functionality, because ultimately I want to support TriggerTemplates but that's been slow progress.

I'll take a look, and I'd like to get this landed.

File: pkg/controller/repository/repository_controller_test.go
Comment (bigkevmcd) [inline pkg/controller/repository/repository_controller_test.go:45]: Can  you maybe gofmt this?

**Review by bigkevmcd — CHANGES_REQUESTED** [review]
Looks good, there's a wee bit of gofmt tidying (and I'll add this to the lint checks) but otherwise, looks good to me, thank you!

File: pkg/controller/repository/repository_controller_test.go
Comment (hnts) [reply pkg/controller/repository/repository_controller_test.go:45]: Thank you for reviewing!

I have already used gofmt.
Should I remove a whitespace before ```=```?

Comment (bigkevmcd) [reply pkg/controller/repository/repository_controller_test.go:45]: @hnts please do, I assumed that gofmt would fix that :-)

After that, I'm happy to merge this.

Comment (hnts) [reply pkg/controller/repository/repository_controller_test.go]: I fixed it!

**Review by bigkevmcd — APPROVED** [review]
Thank you!

**Pull request merged**
